tests/test_meta.py

```diff
@@ -0,0 +1,38 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+import os
+import pkgutil
+import subprocess
+import sys
+import typing
+
+import cryptography
+
+
+def find_all_modules() -> typing.List[str]:
+    return sorted(
+        mod
+        for _, mod, _ in pkgutil.walk_packages(
+            cryptography.__path__,  # type: ignore[attr-defined]
+            prefix=cryptography.__name__ + ".",
+        )
+    )
+
+
+def test_no_circular_imports(subtests):
+    env = os.environ.copy()
+    env["PYTHONPATH"] = os.pathsep.join(sys.path)
+
+    # When using pytest-cov it attempts to instrument subprocesses. This
+    # causes the memleak tests to raise exceptions.
+    # we don't need coverage so we remove the env vars.
+    env.pop("COV_CORE_CONFIG", None)
+    env.pop("COV_CORE_DATAFILE", None)
+    env.pop("COV_CORE_SOURCE", None)
+
+    for module in find_all_modules():
+        with subtests.test():
```

Comment:
Have you considered using `@pytest.mark.parametrize`? Using it allows to load-balance the subtests across workers with plugins like `xdist`. Having a loop in the test itself will probably make the testing less responsive.

Reply:
It's a trade off: parameterize allows splitting with xdist, however it also makes collection more expensive. Because collection is done by each process you can end up with _worse_ performance. In general we've moved away from parameterize for this reason.

In any event, we have enough tests that this happening entirely serially doesn't matter, it just runs in parallel with a test of RSA or something. Given enough cores this would matter again, but we don't run our CI on a threadripper :-(